src/wazuh_db/wdb_global.c

```diff
@@ -1295,9 +1293,75 @@ wdbc_result wdb_global_unassign_agent_group(wdb_t *wdb, int id, cJSON* j_groups)
     return result;
 }
 
+int wdb_global_groups_number_get(wdb_t *wdb, int agent_id) {
+    sqlite3_stmt *stmt = wdb_init_stmt_in_cache(wdb, WDB_STMT_GLOBAL_AGENT_GROUPS_NUMBER_GET);
+
+    if (stmt == NULL) {
+        return OS_INVALID;
+    }
+
+    if (sqlite3_bind_int(stmt, 1, agent_id) != SQLITE_OK) {
+        merror("DB(%s) sqlite3_bind_int(): %s", wdb->id, sqlite3_errmsg(wdb->db));
+        return OS_INVALID;
+    }
+
+    int groups_number = OS_INVALID;
+    cJSON* j_result = wdb_exec_stmt(stmt);
+
+    if (j_result) {
+        if (j_result->child && j_result->child->child) {
+            cJSON* j_groups_number = j_result->child->child;
+            groups_number = j_groups_number->valueint;
+        }
+        cJSON_Delete(j_result);
+    } else {
+        mdebug1("wdb_exec_stmt(): %s", sqlite3_errmsg(wdb->db));
+    }
+
+    return groups_number;
+}
+
+wdbc_result wdb_global_validate_groups(wdb_t *wdb, cJSON *j_groups, int agent_id) {
+    cJSON* j_group_name = NULL;
+    wdbc_result ret = WDBC_OK;
+    int groups_number = 0;
+
+    /* Returns the existent groups number for the agent_id */
+    int existent_groups_number = wdb_global_groups_number_get(wdb, agent_id);
+
+    if (existent_groups_number != OS_INVALID) {
+        cJSON_ArrayForEach (j_group_name, j_groups) {
+            if (cJSON_IsString(j_group_name)) {
+                ++groups_number;
+                if ((groups_number + existent_groups_number) > MAX_GROUPS_PER_MULTIGROUP) {
+                    merror("Invalid groups number. Groups exceed maximum number (%d) permitted", MAX_GROUPS_PER_MULTIGROUP);
```

Comment:
Can we discuss with the team if these `merror` messages should be `mwarn` ?

Reply:
Changed!